### PR TITLE
set text/plain content type when uploading index.txt files

### PR DESCRIPTION
**Pull request by mikehelmick**
## Proposed Changes

* Allow for setting of content type on storage.CreateObject
* Set `text/plain` as the content type for index.txt files

**Release Note**

```release-note
Set content type for index files on blob storage.
```

### CODE REVIEW

**Review by gurayAlsac — APPROVED** [review]
/lgtm
/approve

File: internal/storage/google_cloud_storage.go [hunk 1]
```diff
@@ -44,14 +44,17 @@ func NewGoogleCloudStorage(ctx context.Context) (Blobstore, error) {
...
 // CreateObject creates a new cloud storage object or overwrites an existing one.
-func (s *GoogleCloudStorage) CreateObject(ctx context.Context, bucket, objectName string, contents []byte, cacheable bool) error {
+func (s *GoogleCloudStorage) CreateObject(ctx context.Context, bucket, objectName string, contents []byte, cacheable bool, contentType string) error {
 	cacheControl := "public, max-age=86400"
 	if !cacheable {
 		cacheControl = "no-cache, max-age=0"
 	}
 
 	wc := s.client.Bucket(bucket).Object(objectName).NewWriter(ctx)
 	wc.CacheControl = cacheControl
```
Comment (icco) [inline internal/storage/google_cloud_storage.go:54]: Should we be doing something with https://golang.org/pkg/mime/#TypeByExtension ?

Comment (sethvargo) [reply internal/storage/google_cloud_storage.go:54]: Meh - there's only two types, so I think this is fine for now

**Review by sethvargo — APPROVED** [review]
/lgtm

**Pull request merged**